### PR TITLE
Refactor `object_usage_linter()` to use `xml_nodes_to_lint()`

### LINKED ISSUE #1284: Refactor `object_usage_linter()` to use `xml_nodes_to_lint()`

### PR DESCRIPTION
**Pull request by AshesITR**
fixes #1284 

- remove get_function_assignments()
- use more robust location matching via XPath
- clean up (CC  is 8 now)
- rewrite `get_content()` to accept `xml_node` as info input and use line info as well
- move usage parse adjustments to `parse_check_usage()`
- improve `get_assignment_symbols()` by using `get_r_string()` instead of hand-rolled unescaping

### CODE REVIEW

**Comment by AshesITR** [comment]
Need a news bullet for improved location info in #834 once merged.

Not happy with the performance atm. Looking into it.

**Comment by AshesITR** [comment]
Improvements halves the time, but still at 14.5s vs. 5.7s for main on tools/QC.R.
Willing to sacrifice some performance for the improved location info, but not that much.

File: R/utils.R [hunk 1]
```diff
@@ -110,6 +110,13 @@ get_content <- function(lines, info) {
   lines[is.na(lines)] <- ""
 
   if (!missing(info)) {
+    if (inherits(info, "xml_node")) {
+      info <- lapply(setNames(nm = c("col1", "col2", "line1", "line2")), function(attr) {
+        as.integer(xml2::xml_attr(info, attr))
```
Comment (MichaelChirico) [inline R/utils.R:115]: xp_find_location?

Comment (MichaelChirico) [reply R/utils.R:115]: NVM

Comment (AshesITR) [reply R/utils.R:115]: hmm would be more verbose here.

**Review by MichaelChirico — APPROVED** [review]

**Pull request merged**